optuna/visualization/_pareto_front.py

```diff
@@ -171,53 +130,200 @@ def _make_scatter_object(
     if constraints_func is None:
         data = [
             _make_scatter_object(
-                non_best_trials,
+                info.non_best_trials_with_values,
                 hovertemplate="%{text}<extra>Trial</extra>",
                 dominated_trials=True,
             ),
             _make_scatter_object(
-                best_trials,
+                info.best_trials_with_values,
                 hovertemplate="%{text}<extra>Best Trial</extra>",
                 dominated_trials=False,
             ),
         ]
     else:
         data = [
             _make_scatter_object(
-                infeasible_trials,
+                info.infeasible_trials_with_values,
                 hovertemplate="%{text}<extra>Infeasible Trial</extra>",
                 infeasible=True,
             ),
             _make_scatter_object(
-                non_best_trials,
+                info.non_best_trials_with_values,
                 hovertemplate="%{text}<extra>Feasible Trial</extra>",
                 dominated_trials=True,
             ),
             _make_scatter_object(
-                best_trials,
+                info.best_trials_with_values,
                 hovertemplate="%{text}<extra>Best Trial</extra>",
                 dominated_trials=False,
             ),
         ]
 
-    if n_dim == 2:
+    if info.n_targets == 2:
         layout = go.Layout(
             title="Pareto-front Plot",
-            xaxis_title=target_names[axis_order[0]],
-            yaxis_title=target_names[axis_order[1]],
+            xaxis_title=info.target_names[info.axis_order[0]],
+            yaxis_title=info.target_names[info.axis_order[1]],
         )
     else:
         layout = go.Layout(
             title="Pareto-front Plot",
             scene={
-                "xaxis_title": target_names[axis_order[0]],
-                "yaxis_title": target_names[axis_order[1]],
-                "zaxis_title": target_names[axis_order[2]],
+                "xaxis_title": info.target_names[info.axis_order[0]],
+                "yaxis_title": info.target_names[info.axis_order[1]],
+                "zaxis_title": info.target_names[info.axis_order[2]],
             },
         )
     return go.Figure(data=data, layout=layout)
 
 
+def _get_pareto_front_info(
+    study: Study,
+    target_names: Optional[List[str]] = None,
+    include_dominated_trials: bool = True,
+    axis_order: Optional[List[int]] = None,
+    constraints_func: Optional[Callable[[FrozenTrial], Sequence[float]]] = None,
+    targets: Optional[Callable[[FrozenTrial], Sequence[float]]] = None,
+) -> _ParetoFrontInfo:
+    if axis_order is not None:
+        warnings.warn(
+            "`axis_order` has been deprecated in v3.0.0. "
+            "This feature will be removed in v5.0.0. "
+            "See https://github.com/optuna/optuna/releases/tag/v3.0.0.",
+            DeprecationWarning,
+        )
```

Comment:
[memo] After this PR is merged, `axis_order` is also deprecated in plotly backend.